python/ray/dashboard/modules/train/train_head.py

```diff
@@ -92,6 +94,35 @@ async def get_train_runs(self, req: Request) -> Response:
             content_type="application/json",
         )
 
+    async def _add_actor_status(self, train_runs):
+        actor_status_table = {}
+        try:
+            logger.info("Getting all actor info from GCS.")
+            request = gcs_service_pb2.GetAllActorInfoRequest()
+            reply = await self._gcs_actor_info_stub.GetAllActorInfo(request, timeout=5)
+            if reply.status.code == 0:
+                for message in reply.actor_table_data:
+                    actor_table_data = actor_table_data_to_dict(message)
+                    actor_status_table[actor_table_data["actorId"]] = actor_table_data[
+                        "state"
+                    ]
+        except Exception:
+            logger.exception("Error Getting all actor info from GCS.")
+
+        for train_run in train_runs.values():
+            for worker_info in train_run.workers:
+                worker_info.status = actor_status_table.get(worker_info.actor_id, None)
+
+            from ray.train._internal.state.schema import ActorStatusEnum, RunStatusEnum
+
+            # If any worker died but the run status is not updated, mark it as aborted
+            if train_run.run_status == RunStatusEnum.STARTED:
```

Comment:
let's also check the controller actor?

is this the case. If one worker dies, the entire run is aborted? Will this behavior change in the future. If so, it may be safer to only check the controller actor, assuming when the run aborts, it kills the controller actor.